Lib/encodings/idna.py

```diff
@@ -168,11 +182,13 @@ def encode(self, input, errors='strict'):
         else:
             # ASCII name: fast path
             labels = result.split(b'.')
+            index = 0
             for label in labels[:-1]:
                 if not (0 < len(label) < 64):
-                    raise UnicodeError("label empty or too long")
+                    raise UnicodeEncodeError("idna", input, index, index+len(label), "label empty or too long")
```

Comment:
```suggestion
                    raise UnicodeEncodeError(
                        "idna", input, index, index+len(label),
                        "label empty or too long")
```

Reply:
Agree we using absolute offset.

But it still reports weird range for empty label. I think that it is better to include an opening or closing dot.

Also, I think that it is better to use different messages for empty and too long labels.

Reply:
Split out the empty vs too long cases. Note that `UnicodeEncodeError` takes a python-style [start, end) range, but outputs a (start)-(end-1) inclusive range in the error message, so using something like `0, 0,` gives the weird-looking `0--1` error message whereas `0, 1,` gives `position 0`.